### PR TITLE
[Feat] Defaulting service and workload configuration + Switched to new Option feature

### PR DESCRIPTION
**Pull request by anirudhprasad-sap**
1. Added default configuration for services - xsuaa, saas-registry, service-manager
2. Defaulting workload - Router and CAP
3. Switched to new Options feature
4. Minor fixes

### CODE REVIEW

File: files/xsuaa.yaml.hbs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    xsuaa:
+        name: {{appName}}-uaa
+        serviceOfferingName: xsuaa
+        servicePlanName: broker
+        parameters:
+            xsappname: {{appName}}
+            tenant-mode: shared
+            oauth2-configuration:
+                credential-types:
+                    - instance-secret
```
Comment (anirudhprasad-sap) [inline files/xsuaa.yaml.hbs:11]: @Pavan-SAP Should we default instance-secret?

**Review by Pavan-SAP — APPROVED** [review]
✅

File: files/xsuaa.yaml.hbs (hunk 1 above)
Comment (Pavan-SAP) [reply files/xsuaa.yaml.hbs:11]: No, this does not make sense. Let's only keep binding-secret and perhaps consider adding x509.
Consumers can also override this right?

Comment (anirudhprasad-sap) [reply files/xsuaa.yaml.hbs:11]: Yeah, they can always override it in the values file.

**Pull request merged**